chichiwang/exercise-js
# Build Documentation page sections: Contents, Starting a Project

## Description
Linked to Issue: #65 
Add sections to `Documentation` page: "Contents" and "Starting a Project". In doing so, tweak the `ChecklistItem` component so that the label does not span the entire line, responds properly to screen width, and so that the checkbox animation feels snappier.

## Changes
* Update `app/var.module.scss` to add a property `font-size` to the map `$content`
* Revision pass to `docs/README.md`
  * Add punctuation to section "Starting a Project" line items
  * Split up, and adjust language of, a line item for improved clarity
* Update `pages/Documentation/index.tsx`
  * Wrap page title in a `HeadingBlock` component
  * Create a "Contents" section
  * Create a "Starting a Project" section, using the new `ChecklistItem` component
* Update the `pages/components/ChecklistItem/index.tsx` component
  * Accept a `className` prop to apply to the root element
  * Wrap the entire component in a `div` so that `label` is not the root element (the entire block should not be clickable, just the label contents)
  * Tweak the styles to work well at low screen widths
  * Tweak the animation of the checkbox to feel a bit snappier
* Update `pages/Layout/index.tsx`
  * Apply a new global font-size to the `.content` element

## Steps to QA
* Pull down and switch to this branch
* Run the app and verify in browser that the `/` home route still looks correct (global font-size change)
* Verify in browser that the `/documentation` page looks correct
  * Try at various resolutions
  * Verify the checkboxes behave correctly
* Verify the `docs/README.md` file changes read well
